rolling-shutter/medley/configuration/command/options.go

```diff
@@ -0,0 +1,41 @@
+package command
+
+import "github.com/spf13/afero"
+
+// CommandName overwrites the CLI invokation name of the built command.
+// If this option is not provided, the name returned by the Config struct's Name()
+// method will be used.
+func CommandName(name string) Option {
+	return func(c *commandBuilderConfig) {
+		c.name = name
+	}
+}
+
+// Usage sets the short and long usage strings that are shown in the CLI help messages.
+func Usage(short, long string) Option {
+	return func(c *commandBuilderConfig) {
+		c.shortUsage = short
+		c.longUsage = long
+	}
+}
+
+// WithGenerateConfigSubcommand attaches an additional subcommand
+// 'generate-config' to the command.
+// This allows to generate an example configuration file based on the
+// example configuration parameters provided by the Config structs.
+func WithGenerateConfigSubcommand() Option {
```

Comment:
That's true, I am also not 100% happy with that.
However, in order to pass the init-db functions with an option (`With...`), you either:
- have to make the whole option system generic, and then you need to provide all other options with a way to infer the type. This will also be weird to call
- or the `WithInitDBCommand` option would have to use an `(initDbFunc func(configuration.Config))` call signature and then cast that to the concrete type at runtime, omitting the compile time type safety. Since we have the smoke-test errors here would be caught during development, so this is actually my preferred option out of the two.

Do you have another idea?  Do you think it's worth implementing this right now?

Reply:
Yeah true that makes it more complicated.

No I’d definitely not hold this PR up over this.

Let’s make an issue and get back to it later.